### DISCUSSION TITLE
Date from which on scripts are available

### DISCUSSION THREAD

**Discussion by StukovTTV** [TVTower/TVTower#847] [Ideas]
One idea I had:
with movies we have a date when they are released, but scripts do not have it as far as i see. Every Script is probably available from gamestart.
Wouldn't it be nice if we could also say something like Script A is available from year 2020 onwards?

Another idea, but less relevant, is - for script B being available somebody should have produced Script A. Like Compilation 2 is only available if somebody allready produced Compilation 1

**Comment by nittka** [comment]
Script templates can have `availability` as child node (https://github.com/TVTower/Documentation/blob/master/database_de/scripts.md#kindelemente-von-scripttemplate), so it is already possible define restrictions. However, I have not checked if any existing template actually uses that option.

With respect to restrictions on what can be produced, I would not start on the "script" side. In the best case, they really are templates with many variations. It would not make much sense to restrict the production of one generic adventure script because anoter generic adventure script has not yet been produced. Also note that the scripts available in the agency are the same for all players.

To my mind, a more realistic approach would be that production (some) companies and celebrities will not work for you (or only for a much higher price until you reach a certain channel sympathy), if you did not prove yourself with previous productions. You as a producer must gain experience before you are trusted "with the good stuff". This way, you cannot simply start of with a great script (which would be wasted on a "cheap" production), but have to work your way upwards.

**Comment by StukovTTV** [comment]
I totally missed the possibility of the availability node. Sorry
That is actually great, and with that i can also simulate the II not before I if i say Godmother I can be produced from 1960 to 1990 and Godmother II can be produced from 1991 onwards i can never produce I after II :)

In scripts i am planning to have real nice films available too. Also the "neutral" studios will produce films from the scripts too which is nice imo as then there will always be new films even if you start somehow in 2040 or so :)

**Reply by nittka** [reply]
What you define in the database is actually a script template, i.e. the basis for many scripts. These templates are most "effective" if they are generic and can be used over and over (using variables for name and description to prevent repetitions).
The advantage of a "real move" as script (cool, I can produce that) has to be weighed against the frustration (why can I produce the same movie 50 times) or the effort of creating a one-time script (if it was removed after the first production.

**Reply by StukovTTV** [reply]
Interesting, i totally missed that a scripttemplate will be automaticly available multiple time without andy scriptflags. Well I guess there are tons of remakes in the real world, or i can really make them single-productions

**Reply by StukovTTV** [reply]
I just thought about I can use real Movies and Variables too 
Die Bourne Identität
Das Bourne Vermächtnis
Jason Bourne
Die Bourne Verschwöring
Das Bourne Vermächtnis

Imediatly I would have covered several films
Of course I need to have a more general Description, but it would be the best of two worlds. Of course it still will happen to have duplicate names

**Comment by GWRon** [comment]
Ich wuerde einige Dinge nicht davon abhaengig machen ob das vorherige Drehbuch erstellt wurde.

Wichtiger waere es ja, ein Drehbuch erst nach Fertigstellung der Produktion verfuegbar zu machen (auch nicht 100% korrekt, aber verhindert es, dass dann beide Drehbuecher ungenutzt beim Makler auftauchen).


Auch sollten solche Drehbuecher die Bearbeitbarkeit der Titel/Beschreibung einschraenken. Sonst erzeugt es den "Nachfolger zum Hit", den es gar nicht gibt.

Das macht es naemlich schon ein ganzes Stueck komplizierter.


Allerdings verstehe ich natuerlich den Reiz, bekannte Filme selbst herzustellen.

**Comment by nittka** [comment]
I don't see a real reason to offer scripts for (individual) real movies. If it is important for you to be able to produce them, simply take the script of a generic movie for the genre and change title and description.

**Comment by GWRon** [comment]
I think these changes should be done in a separate xml. Doing that ingame might be.. tedious already the second game you want to do that movie.

**Reply by nittka** [reply]
True, but at least that way, you can produce any movie you like in any game. The average player will not edit xml files. And re-loading database files is possible only in dev mode and is not well tested (definitely buggy for series episodes).

**Reply by GWRon** [reply]
Was not talking about reloading files - I just thought about separating these changes into an extra file which you can use or not use in your game.

Anyways it could be "doable" to have an "appenddb(file)" which does not override stuff but adds new elements not existing yet. Yet not sure if it is useful

**Comment by StukovTTV** [comment]
As availability is allready possible, i am pretty happy what i can do. I can set real movies to be produced, i juest need to set it how i want it. For most of the movies i selected i don't even need an availability tag.

Another benefit for me with the movies in scripts is ... i can start in let's say 2045 and increase the neutral producer companies and so still get a few new films every year. Of course that can be made with only fictional films too.
